### PR TITLE
Add Dispose and IsConnected to IDataQueueHandler

### LINKED ISSUE #4348: Add Dispose and IsConnected to IDataQueueHandler
#### Expected Behavior
- `IDataQueueHandler` implementations should be able to perform cleanup (e.g. closing  connections) and also have a method to test for connectivity.

#### Actual Behavior
- `IDataQueueHandler` does not have `Dispose` and `IsConnected` methods

#### Potential Solution
- Make `IDataQueueHandler` inherit from `IDisposable`
- Add `IDataQueueHandler.IsConnected`

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

### PR DESCRIPTION
**Pull request by StefanoRaggi**
#### Description
- The `IDataQueueHandler` interface has been extended to include `Dispose` and `IsConnected` methods

#### Related Issue
Closes #4348 

#### Motivation and Context
- `IDataQueueHandler` implementations should be able to perform cleanup (e.g. closing connections) and also have a method to test for connectivity.

#### Requires Documentation Change
No.

#### How Has This Been Tested?
- Deployed live algorithms in QC cloud

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Engine/DataFeeds/Queues/FakeDataQueue.cs [hunk 1]
```diff
@@ -120,6 +120,19 @@ public void Unsubscribe(LiveNodePacket job, IEnumerable<Symbol> symbols)
...
+        /// </summary>
+        /// <returns>true if the data provider is connected</returns>
+        public bool IsConnected => true;
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
+        {
+        }
```
Comment (Martin-Molinero) [inline Engine/DataFeeds/Queues/FakeDataQueue.cs:134]: nit - FakeDataQueue actually has a `Timer` to dispose of

File: ToolBox/IQFeed/IQFeedDataQueueHandler.cs [hunk 2]
```diff
@@ -332,6 +329,13 @@ public bool CanAdvanceTime(SecurityType securityType)
         {
             return _symbolUniverse.CanAdvanceTime(securityType);
         }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
```
Comment (Martin-Molinero) [inline ToolBox/IQFeed/IQFeedDataQueueHandler.cs:336]: IQFeedDQH seems to be missing a lot of Dispose calls and some refactoring to achieve it too, maybe we can generate a new issue just for this?

**Comment by Martin-Molinero** [comment]
I think `LiveTradingDataFeed` should now probably be calling `Dispose` on the `IDataQueueHandler` it owns?

File: Engine/DataFeeds/Queues/FakeDataQueue.cs (hunk 1 above)
Comment (StefanoRaggi) [reply Engine/DataFeeds/Queues/FakeDataQueue.cs:134]: Will update

File: ToolBox/IQFeed/IQFeedDataQueueHandler.cs (hunk 2 above)
Comment (StefanoRaggi) [reply ToolBox/IQFeed/IQFeedDataQueueHandler.cs:336]: Sure, we can create a new issue when this one is merged 👍

**Comment by StefanoRaggi** [comment]
> I think `LiveTradingDataFeed` should now probably be calling `Dispose` on the `IDataQueueHandler` it owns?

This might be tricky because the `IDataQueueHandler` instance is usually the brokerage instance.

**Pull request merged**